### PR TITLE
Indexing and aggregation pipeline optimisation

### PR DESCRIPTION
**Pull request by SianaDicheva**
# Description

Changes in several files to accommodate the addition of database indexes and optimisation of the aggregation pipeline. 

## Useful links

- Ticket number - 68884

## Type of change

- [ ] Improvement (refactor or addition to existing functionality)

## Reference document
[Indexing.md](https://github.com/ReliefApplications/ems-backend/files/13299356/Indexing.md)

### CODE REVIEW

**Comment by AntoineRelief** [comment]
@SianaDicheva 
Quick question for this one
Is there a minimum string size for the search to apply?

You can see with the gif below ( tested in your environment > Query Optimisation > Filtered events )
I was able to replicate the issue on other environments


https://github.com/ReliefApplications/ems-backend/assets/50317271/78f5318d-f804-48bb-9ece-1d9ca595bcf1

**Comment by AntoineRelief** [comment]
@SianaDicheva @morgang-adappt 
I found two other issues while testing the feature in our UAT environment:

- first one is about filtering on id fields:
https://www.loom.com/share/03eb0b02284c4890a4ebc80e30a1681c?sid=3953c8a3-852d-403f-9996-d61fc6dd54f8
- second one is about filtering on calculated fields:
https://www.loom.com/share/03ec3e3db4744c22b303ae3d8adf3e98?sid=908cdfb7-8089-446e-810b-1484e39e32d4

**Comment by SianaDicheva** [comment]
> @SianaDicheva @morgang-adappt I found two other issues while testing the feature in our UAT environment:
> 
> * first one is about filtering on id fields:
>   https://www.loom.com/share/03eb0b02284c4890a4ebc80e30a1681c?sid=3953c8a3-852d-403f-9996-d61fc6dd54f8
> * second one is about filtering on calculated fields:
>   https://www.loom.com/share/03ec3e3db4744c22b303ae3d8adf3e98?sid=908cdfb7-8089-446e-810b-1484e39e32d4

Thank you very much for providing these issues as well! We will take them into consideration and be part of the changes, which we are currently developing. The first one we have fixed locally and working on second one!

File: package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
     "graphql-subscriptions": "^2.0.0",
     "graphql-type-json": "^0.3.2",
     "graphql-ws": "^5.14.0",
+    "helmet": "^7.0.0",
```
Comment (AntoineRelief) [inline package.json:70]: to be removed during review

File: src/schema/mutation/editResource.mutation.ts [hunk 2]
```diff
@@ -707,25 +706,6 @@ export default {
       if (args.calculatedField) {
         const calculatedField: CalculatedFieldChange = args.calculatedField;
 
-        // Check if calculated field expression is too long
-        if (calculatedField.add || calculatedField.update) {
```
Comment (AntoineRelief) [inline src/schema/mutation/editResource.mutation.ts:711]: can be removed as it should not raise any error
to be tested during review

File: src/utils/schema/resolvers/Query/all.ts [hunk 3]
```diff
@@ -672,6 +700,16 @@ export default (entityName: string, fieldsByName: any, idsByName: any) =>
         }
       }
 
+      const rawDataProjection = (data: any) => {
```
Comment (AntoineRelief) [inline src/utils/schema/resolvers/Query/all.ts:703]: can be removed during review

**Pull request merged**